omod/src/main/java/org/openmrs/tag/web/rest/resource/TagResource.java

```diff
@@ -87,7 +87,11 @@ public Tag getByUniqueId(String uuid) {
 	 */
 	@Override
 	public Tag save(Tag tag) {
-		return getService().saveTag(tag);
+		try {
+			return getService().saveTag(tag);
+		}
+		catch (IllegalArgumentException e) {}
+		return null;
```

Comment:
Don't catch the exception saveTag, the caller of this method should take care of it

Reply:
It means, you can just remove these changes

Reply:
the saveTag method throws an exception in case of duplicates. If we don't catch it in the resource, it gets reflected in the UI.
Another way out would be to log a warning instead of throwing an exception in the saveTag() method itself

Reply:
Sorry, I just realized this is in the resource, I thought it was in the service, I guess this is fine.

Reply:
On the other hand, I think the proper thing to do is to not catch the exception here, this swallows the error, I think you should let the exception bubble, in the UI you want to check if the object doesn't yet have the tag before posting it to the server.

Reply:
Ok. so do  you want me to handle this exception in the UI. Do you want the error message, or should i not display anything when a duplicate tag is being added ?

Reply:
That's not what am saying, what ma suggesting is that when the user submits the new tag, on the client side you need to first check if the domain object already has the tag and if they do, do nothing or you might show them a message that says they already have the tag.

Reply:
Note that am saying you need to check on the client side if they already have the tag before posting the save request